contrib/openshift/osba-os-template.yaml

```diff
@@ -0,0 +1,238 @@
+kind: Template
+apiVersion: v1
+objects:
+- kind: Secret
+  apiVersion: v1
+  metadata:
+    name: osba-redis
+    labels:
+      app: osba-redis
+      release: "osba"
+  type: Opaque
+  data:
+    redis-password: "cGFzc3dvcmQ="
+- kind: Secret
+  apiVersion: v1
+  metadata:
+    name: osba-open-service-broker-azure-auth
+    labels:
+      app: osba-open-service-broker-azure
+      release: "osba"
+  type: Opaque
+  data:
+    username: dXNlcm5hbWU=
+    password: cGFzc3dvcmQ=
+- kind: Secret
+  apiVersion: v1
+  metadata:
+    name: osba-open-service-broker-azure
+    labels:
+      app: osba-open-service-broker-azure
+      release: "osba"
+  stringData:
+    azure-client-secret: ${AZURE_CLIENT_SECRET}
+    encryption-key: This is a key that is 256 bits!!
+    redis-password: "password"
+    basic-auth-password: "password"
+- kind: Service
+  apiVersion: v1
+  metadata:
+    name: osba-redis
+    labels:
+      app: osba-redis
+      release: "osba"
+  spec:
+    ports:
+    - name: redis
+      port: 6379
+      targetPort: redis
+    selector:
+      app: osba-redis
+- kind: Service
+  apiVersion: v1
+  metadata:
+    name: osba-open-service-broker-azure
+    labels:
+      app: osba-open-service-broker-azure
+      release: "osba"
+  spec:
+    type: ClusterIP
+    selector:
+      app: osba-open-service-broker-azure
+    ports:
+    - name: http
+      protocol: TCP
+      port: 80
+      targetPort: 8080
+
+- apiVersion: v1
+  kind: PersistentVolumeClaim
```

Comment:
This resource should be created conditionally, depending on whether one wants to use on-cluster Redis or a remote Redis (i.e. Azure Redis Cache).